src/state/hooks.ts

```diff
@@ -78,13 +78,54 @@ export const useFarmUser = (pid) => {
   }
 }
 
-export const useLpTokenPrice = (symbol: string) => {
-  const farm = useFarmFromSymbol(symbol)
-  const tokenPriceInUsd = useGetApiPrice(getAddress(farm.token.address))
+export const useFarmFromTokenSymbol = (tokenSymbol: string, preferredQuoteTokens?: string[]): Farm => {
+  const farms = useSelector((state: State) => state.farms.data.filter((f) => f.token.symbol === tokenSymbol))
+  const filteredFarm = filterFarmsByQuoteToken(farms, preferredQuoteTokens)
+  return filteredFarm
```

Comment:
Filter farm to get a 'preferred' LP, for example if we were to call the function with tokenSymbol `KUN`, two farms would match the filter:

```ts
  {
    pid: 373,
    lpSymbol: 'KUN-BUSD LP',
    token: tokens.kun,
    quoteToken: tokens.busd,
  },
  {
    pid: 372,
    lpSymbol: 'KUN-QSD LP',
    token: tokens.kun,
    quoteToken: tokens.qsd,
  },
```

We would want to return the `BUSD` farm to get the token price, as it is the most straightforward conversion (in fact in this instance, we'd have no way of getting price from `QSD` as there is no `QSD - someQuoteToken` pair).

The actual `filteredFarm` logic is within `'utils/farmsPriceHelpers'`

Reply:
Please add a comment to this effect (shortened of course).